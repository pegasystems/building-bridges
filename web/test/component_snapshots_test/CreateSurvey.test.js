import React from 'react';
import CreateSurvey from '../../src/Components/SurveyCreation';
import renderer from 'react-test-renderer';

global.fetch = jest.fn(() =>
    Promise.resolve({
        json: () => Promise.resolve([{
            date: '2020-07-03',
            hideVotes: true,
            key: 'title-1',
            secret: 'cf9ff3424c90759a',
            title: 'title'
        },
        {
            date: '2020-07-04',
            hideVotes: false,
            key: 'title-2',
            secret: '89as8n97asdfkdf',
            title: 'title-2'
        }])
    })
);


test('CreateSurvey renders properly test', () => {
    const component = renderer.create(
        <CreateSurvey/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


