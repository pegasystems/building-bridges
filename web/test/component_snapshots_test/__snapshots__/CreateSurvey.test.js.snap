// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateSurvey renders properly test 1`] = `
<div
  className="newSurveys"
>
  <div
    className="new-survey-box"
  >
    <form
      id="createBridge"
      onSubmit={[Function]}
    >
      <h4>
        Survey name
      </h4>
      <div
        className="creation-flex"
      >
        <input
          className="survey-name"
          onChange={[Function]}
          placeholder="Title of My Bridges Survey"
          type="text"
          value=""
        />
      </div>
      <h4>
        Optional Features
      </h4>
      <div
        className="options"
      >
        <label
          className="container-checkbox"
          htmlFor="hide_votes"
        >
          Hide vote totals until after a user votes
          <input
            id="hide_votes"
            name="hide_votes"
            onChange={[Function]}
            type="checkbox"
            value="false"
          />
          <span
            className="checkmark"
          />
        </label>
      </div>
      <input
        className="add-new-button"
        type="submit"
        value="Create survey"
      />
    </form>
  </div>
</div>
`;
